home/tests/test_explore_page.py

```diff
@@ -20,17 +28,61 @@ def test_answers_num_func(self, question_id, expected_result):
 
     class TestExplorePageView:
 
-        @pytest.mark.django_db
-        @pytest.fixture
-        def explore_page_response(self, client):
-            url = reverse('explore-page')
-            response = client.get(url)
-            return response
-
         @pytest.mark.django_db
         def test_explore_page_response(self, explore_page_response):
             assert explore_page_response.status_code == 200
             assert explore_page_response.templates[0].name == "home/explore.html"
 
             questions = Question.objects.all()
             assert set(explore_page_response.context['questions']) == set(questions)
+
+    class TestTagsFilter:
+
+        @pytest.mark.parametrize(('tag_name', 'included_question', 'not_included_question'), [
+            ("Bagrut_Exam", 'Question #2', 'Question #10'),
+            ("5th_Grade", 'Question #4', 'Question #5'),
+            ("Pitagoras", 'Question #1', 'Question #2'),
+        ])
+        @pytest.mark.django_db
+        def test_questions_filter_by_tag(self, client, tag_name, included_question, not_included_question):
+            url = f'/explore/?tag={tag_name}'
+            response = client.get(url)
+
+            requested_questions = response.context['questions']
+            assert included_question in str(requested_questions)
```

Comment:
Why do string comparison when you can compare actual lists or IDs? 

The Python `in` operator is not just for strings, but an `in` test  is generally weak, because its ignoring unwanted results.